sections/testingandquality/refactoring.md

```diff
@@ -0,0 +1,27 @@
+# Refactoring
+
+<br/><br/>
+
+### Motivation for Refactoring
+
+Refactoring is an important process in the iterative development flow. Removing "Code Smells" (bad coding practices) such as Duplicated Code, Long Methods, Long Parameter list will improve your code and making it more maintainable. Using a static analysis tools will assist you in finding these code smells and build a process around refactoring.
```

Comment:
@TheHollidayInn I find it important to say a sentence or two about (1) the process - how it's used: code review, CI, within IDE. "Code smells are a great candidate to start a code review process from" (2) tangible features - "from straightforward features like detecting long functions to modern tools that can detect in real-time code paths that are not tested"